src/file_io.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import numpy
+import re
+import pickle
+
+from . import graphs, plotting_tools
+from .data import Data
+
+def save_project(self, path):
```

Comment:
Yeah I was already considering moving this to it's own function, as it didn't have a logical place anywhere else.